repo2docker/buildpacks/r.py

```diff
@@ -64,21 +71,22 @@ def detect(self):
         unless a `requirements.txt` is present and we do not want to require the
         presence of a `requirements.txt` to use R.
 
-        Instead we just check if runtime.txt contains a string of the form
-        `r-<YYYY>-<MM>-<DD>`
+        Instead we check the options described in the class comment above.
         """
         # If no date is found, then self.checkpoint_date will be False
         # Otherwise, it'll be a date object, which will evaluate to True
         if self.checkpoint_date:
             return True
+
         description_R = 'DESCRIPTION'
-        if not os.path.exists('binder') and os.path.exists(description_R):
-            if not self.checkpoint_date:
-                # no R snapshot date set through runtime.txt
-                # set the R runtime to the latest date that is guaranteed to be on MRAN across timezones
-                self._checkpoint_date = datetime.date.today() - datetime.timedelta(days=2)
-                self._runtime = "r-{}".format(str(self._checkpoint_date))
-            return True
+        if not os.path.exists('binder'): 
+            if os.path.exists(description_R) or (self.stencila_manifest_contexts and "r" in self.stencila_manifest_contexts):
```

Comment:
Can we make `self.stencila_manifest_contexts` default to an empty list/set instead of `None` and then change this to:

```suggestion
            if os.path.exists(description_R) or "r" in self.stencila_manifest_contexts:
```

Reply:
We sure can! I just learned that empty lists are falsy, and other checks just do `if self.stencila_manifests_contexts`.

Reply:
They are indeed false-y, but you can check membership of an empty list so we don't have to check first if it is empty/false-y or not.